lib/pbench/test/unit/server/query_apis/test_datasets_detail.py

```diff
@@ -30,25 +31,37 @@ def _setup(self, client):
 
     api_method = API_METHOD.GET
 
+    def token(self, client, config: PbenchServerConfig, user: str) -> str:
+        response = client.post(
+            f"{config.rest_uri}/login",
+            json={"username": user, "password": "12345"},
+        )
+        assert response.status_code == HTTPStatus.OK
+        data = response.json
+        assert data["auth_token"]
+        return data["auth_token"]
```

Comment:
Technically, I don't think that's true since `response.json` is *deserializing* the byte-stream `data` into a `dict` structure; but there's also no reason to look up the `"auth_token"` key twice in this context. (I didn't comment because I recognized this pattern and I think he just copied it from somewhere else... and it's "only a unit test". 😬 )

Reply:
OK, I shall amend:  given that fetching the `json` attribute is only "half" of the job, I would have cached the result instead of the intermediate step.  😀